7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -191,18 +202,25 @@ adjust_lake_cell_tile_xwalk <- function(spatial_xwalk, lake_centroids, query_cel
   cell_centroids_with_data <- query_cell_centroids %>%
     filter(cell_no %in% cells_with_data)
 
-  # match each lake to a cell that returned data based on
-  # which cell centroid is closest to each lake centroid
-  lake_cells_with_data_tiles_xwalk <- lake_centroids %>%
-    st_join(cell_centroids_with_data, join=st_nearest_feature, left=TRUE) %>%
-    st_drop_geometry() %>%
-    select(site_id, state, cell_no, tile_no)
-
-  # join this adjusted xwalk to the spatial xwalk
-  adjusted_xwalk <- spatial_xwalk %>%
-    left_join(lake_cells_with_data_tiles_xwalk,
-              by=c('site_id', 'state'),
-              suffix=c('_spatial','_data'))
+  # match each lake to a cell that returned data, preferably to a cell 1) within the same row as, and
+  # 2) within `x_buffer` x distance (number of columns) of, the cell within which the lake falls spatially
+  # If no non-nan cells meet that criteria, match instead to the closest non-nan cell, regardless of row
+  adjusted_xwalk <- lake_centroids %>%
+    left_join(spatial_xwalk, by=c('site_id', 'state')) %>% # add cell_no, tile_no for each lake
+    left_join(query_cell_centroids %>% st_drop_geometry(), by=c('cell_no','tile_no')) %>% # add x, y coords of cell_no for each lake
+    rename(x_spatial=x, y_spatial=y, cell_no_spatial=cell_no, tile_no_spatial=tile_no) %>%
+    group_by(y_spatial, x_spatial) %>%
+    group_modify(~ {
+      # subset the non-nan cells to those with the same y value (in the same row) as the cell with missing data
+      # and those within the specified x_buffer distance (+- x_buffer columns) from the cell with missing data
+      cells_in_row_and_within_x_buffer <- cell_centroids_with_data %>% filter(y==.y$y_spatial & abs(x-.y$x_spatial) <= x_buffer)
+      # if no non-nan cells meet that criteria, join instead to the nearest non-nan cell regardless of y value
+      cells_to_match <- if (nrow(cells_in_row_and_within_x_buffer) > 0) cells_in_row_and_within_x_buffer else cell_centroids_with_data
+      .x %>% st_join(cells_to_match,
+                     join=st_nearest_feature, left=TRUE)
+    }) %>%
+    ungroup() %>%
+    select(site_id, state, cell_no_spatial, tile_no_spatial, cell_no_data=cell_no, tile_no_data=tile_no)
```

Comment:
Hmmm seeing these column names makes me think this says "no data" and "no spatial". Wonder if we should drop `no` here or write as `num` ? What do you think?

Reply:
ah yes, I see your point. I'm reluctant to change the use of `cell_no` since we use it throughout this pipeline and in `lake_temperature_process_models`. Maybe we could change these to `spatial_cell_no` and `data_cell_no`?

Reply:
oh I think that would help!